src/nuclide_header.F90

```diff
@@ -690,6 +751,72 @@ pure function nuclide_nu(this, E, emission_mode, group) result(nu)
 
   end function nuclide_nu
 
+!===============================================================================
+! COMPUTE_NUCLIDE_NU_PHOTON_TOTAL is an interface to compute the number of
+! photons produced
+!===============================================================================
+
+  pure function compute_nuclide_nu_photon_total(this, E, t, i_log_union) result(nu_photon_total)
+    class(Nuclide), intent(in) :: this
+    real(8),        intent(in) :: E
+    integer,        intent(in) :: t
+    integer,        intent(in) :: i_log_union
+    real(8)                    :: rx_xs
+    real(8)                    :: nu_photon_total
+    real(8)                    :: f
+    integer                    :: m, j, k
+    integer                    :: i_grid, i_low, i_high
+
+    associate (grid => this % grid(t), xs => this % sum_xs(t))
+      ! Determine the energy grid index using a logarithmic mapping to
+      ! reduce the energy range over which a binary search needs to be
+      ! performed
+
+      if (E < grid % energy(1)) then
+        i_grid = 1
+      elseif (E > grid % energy(size(grid % energy))) then
+        i_grid = size(grid % energy) - 1
+      else
+        ! Determine bounding indices based on which equal log-spaced
+        ! interval the energy is in
+        i_low  = grid % grid_index(i_log_union)
+        i_high = grid % grid_index(i_log_union + 1) + 1
+
+        ! Perform binary search over reduced range
+        i_grid = binary_search(grid % energy(i_low:i_high), &
+             i_high - i_low + 1, E) + i_low - 1
+      end if
+
+      ! check for rare case where two energy points are the same
+      if (grid % energy(i_grid) == grid % energy(i_grid + 1)) &
+           i_grid = i_grid + 1
+
+      ! calculate interpolation factor
+      f = (E - grid % energy(i_grid)) / &
+           (grid % energy(i_grid + 1) - grid % energy(i_grid))
+    end associate
+
+    nu_photon_total = ZERO
+
+    ! Calculate nu-photon total cross section
+    do m = 1, size(this % reactions)
+      associate (rx => this % reactions(m))
+        j = rx % xs(t) % threshold
+        do k = 1, size(rx % products)
+          if (rx % products(k) % particle == PHOTON) then
+            if (i_grid >= j) then
+              rx_xs = (ONE - f) * rx % xs(t) % value(i_grid - j + 1) &
+                   + f * rx % xs(t) % value(i_grid - j + 2)
+              nu_photon_total = nu_photon_total + rx_xs * &
+                   rx % products(k) % yield % evaluate(E)
+            end if
+          end if
+        end do
+      end associate
+    end do
+
+  end function compute_nuclide_nu_photon_total
```

Comment:
Related to my other question -- isn't this function redundant since sum_xs already stores nu_photon_total?